design/ceph-csi-operator.md

```diff
@@ -0,0 +1,84 @@
+
+## Upgraded clusters
+
+For upgrade clusters, we'll fetch the csi settings and cluster connections details from configmaps like `rook-ceph-operator-config` and `rook-ceph-csi-config`. After fetching the csi settings and cluster connections
+details we can fill the new ceph-csi operator cr `CephCSICephConfig` with connection details and other CR's. Once we have the settings we can set `USE_CSI_OPERATOR:` is to `true` it will also disable the csi `ROOK_CSI_DISABLE_DRIVER`.
+
+**Note:** Not much changes are anticipated for external mode clusters.
+
+## New Ceph-CSI Operator CRDs
+
+### Ceph-CSI-Opeator configuration
+
+The CSI operator brings new CRDS and API for users to configure settings. These CRDs are:
+
+### CephCSICephCluster CRD
+
+[This CRD](https://github.com/ceph/ceph-csi-operator/blob/main/docs/design/operator.md#cephcsicephcluster-crd) stores connection and configuration details for a single Ceph cluster and provide the information to be used by multiple CSI drivers.
+
+Any setting in this CR either needs to be generated by Rook, or wrapped by CephCluster CR under `spec.csi` or `rook-ceph-cluster` helm chart `values.yaml`.
+
+### CephCSIConfig CRD
+
+[This CRD](https://github.com/ceph/ceph-csi-operator/blob/main/docs/design/operator.md#cephcsiconfig-crd) contains details about CephFS, RBD, and NFS configuration to be used when communicating with Ceph. Include a reference to a CephCSICephCluster holding the connection information for the target Ceph cluster.
+
+The Rook operator would generate an instance of these CR for each of CephFilesystemSubVolumeGroup and CephBlockPoolRadosNamespace CRs.
+
+### CephCSIOperatorConfig CRD
+
+[This CRD](https://github.com/ceph/ceph-csi-operator/blob/main/docs/design/operator.md#cephcsioperatorconfig-crd)  manages operator-level configurations and offers a place to overwrite settings for CSI drivers. This CRD is a namespace-scoped CRD and a single CR named ceph-csi-operator-config can be created by an user inside the operator's namespace.
+
+### CephCSIDriver CRD
+
+Rook will define three instances of [this CRD](https://github.com/ceph/ceph-csi-operator/blob/main/docs/design/operator.md#cephcsidriver-crd), one each for RBD, CephFS, and NFS for managing the installation, lifecycle management, and configuration for CephFS, RBD, and NFS CSI drivers within namespaces.
+The CRs will be empty (all the settings are defined as defaults in the previous CR), or else some small set of default driver-specific settings.
+
+### CephCSIConfigMapping
+
+[This CRD](https://github.com/ceph/ceph-csi-operator/blob/main/docs/design/operator.md#cephcsiconfigmapping) contains a mapping between local and remote Ceph cluster configurations.It also provides a mapping between the CephCSIConfigurations.
+Rook will define these mappings when the rbd mirroring is enabled.
+
+## Open Questions
+
+1. Should automate the deployment of csi-operator internally or leave it to users to deploy via manifests.
```

Comment:
My opinion here is that it makes the most sense to (at least starting out) have a separate `csi.yaml` maifest that deploys the CSI operator and whatever RBAC it needs.

Reply:
Agreed on this one as its experimental feature.